dask/dataframe/core.py

```diff
@@ -1669,7 +1671,11 @@ def mean(self, axis=None, skipna=True, split_every=False, dtype=None, out=None):
             )
             return handle_out(out, result)
         else:
-            num = self._get_numeric_data()
+            dtype = getattr(self, "dtype", None)
+            if dtype == "<M8[ns]":
+                num = self.dropna()._get_numeric_data().astype("i8")
```

Comment:
What happens with `.mean(skipna=False)`? Presumably we shouldn't drop missing values then.

Reply:
Can you add this is a test case?